### PR TITLE
Update Windows installer to accept symmetric keys for DPS

### PR DESCRIPTION
**Pull request by mrohera**
To support Edge installation and initialization using DPS symmetric key attestation, a new optional CLI parameter 'SymmetricKey' was introduced to accept a symmetric key meant to provision the device in its designated IoT Hub via DPS.

The new sample CLI invocations have documented in the script below.

### CODE REVIEW

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 1]
```diff
@@ -71,6 +71,7 @@ PS> Initialize-IoTEdge -Manual -DeviceConnectionString $deviceConnectionString -
 .EXAMPLE
 
 PS> Initialize-IoTEdge -Dps -ScopeId $scopeId -RegistrationId $registrationId -ContainerOs Windows
+PS> Initialize-IoTEdge -Dps -ScopeId $scopeId -RegistrationId $registrationId -ContainerOs Windows -SymmetricKey $symmetricKey
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: This should be a separate `.EXAMPLE`

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 2]
```diff
@@ -337,6 +342,7 @@ PS> Install-IoTEdge -Manual -DeviceConnectionString $deviceConnectionString -Con
 .EXAMPLE
 
 PS> Install-IoTEdge -Dps -ScopeId $scopeId -RegistrationId $registrationId -ContainerOs Windows
+PS> Install-IoTEdge -Dps -ScopeId $scopeId -RegistrationId $registrationId -ContainerOs Windows -SymmetricKey $symmetricKey
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: This should be a separate `.EXAMPLE`

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 3]
```diff
@@ -1303,13 +1314,17 @@ function Set-ProvisioningMode {
...
         else {
-            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*global_endpoint:\s*".*"\s*#?\s*scope_id:\s*".*"\s*#?\s*registration_id:\s".*"'
+            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*global_endpoint:\s*".*"\s*#?\s*scope_id:\s*".*"\s*#?\s*registration_id:\s*".*"\s*#?\s*symmetric_key:\s".*"'
             $replacementContent = @(
                 'provisioning:',
                 '  source: ''dps''',
                 '  global_endpoint: ''https://global.azure-devices-provisioning.net''',
                 "  scope_id: '$ScopeId'",
                 "  registration_id: '$RegistrationId'")
+            if ($NULL -ne $SymmetricKey)
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: `if ($SymmetricKey -ne '')` or `if ($SymmetricKey)` (PS strings are not nullable, and don't use yoda conditionals.)

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 4]
```diff
@@ -95,6 +96,10 @@ function Initialize-IoTEdge {
         [Parameter(Mandatory = $true, ParameterSetName = 'DPS')]
         [String] $RegistrationId,
 
+        # The DPS symmetric key to provision the Edge device identity
+        [Parameter(Mandatory = $false, ParameterSetName = 'DPS')]
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: Don't need to set `Mandatory` to `$false`. It's the default.

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 5]
```diff
@@ -361,6 +367,10 @@ function Install-IoTEdge {
...
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: Don't need to set `Mandatory` to `$false`. It's the default.

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 6]
```diff
@@ -1303,13 +1314,17 @@ function Set-ProvisioningMode {
...
-            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*global_endpoint:\s*".*"\s*#?\s*scope_id:\s*".*"\s*#?\s*registration_id:\s".*"'
+            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*global_endpoint:\s*".*"\s*#?\s*scope_id:\s*".*"\s*#?\s*registration_id:\s*".*"\s*#?\s*symmetric_key:\s".*"'
             $replacementContent = @(
                 'provisioning:',
                 '  source: ''dps''',
                 '  global_endpoint: ''https://global.azure-devices-provisioning.net''',
                 "  scope_id: '$ScopeId'",
                 "  registration_id: '$RegistrationId'")
+            if ($NULL -ne $SymmetricKey)
+            {
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: `{` goes on the same line as the `if`.

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 (hunk 3 above)
Comment (mrohera) [reply scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: I get the null check but with yoda, what beef do you have?

**Pull request merged**